client/tinychain/ml/dnn.py

```diff
@@ -64,15 +197,26 @@ def train(self, inputs, cost):
                 A.append(A_l.copy())
                 Z.append(Z_l)
 
-            m = inputs.shape[0]
+            m = inputs.shape[0]   
             dA = cost(A[-1]).sum() / m
 
             updates = []
+            
+
+            l_m,l_v,t=optimizer['l_m'],optimizer['l_v'],optimizer['t']
```

Comment:
These implementation details of AdamOptimizer need to go somewhere else, outside of `DNN.train`, so that it can accept other optimizer algorithms as well